### PR TITLE
Made test warnings fatal

### LINKED ISSUE #1467: GitHub Actions not failing on warnings
**Describe the bug**
Our `Codebase testing` section of our [pull-request.yml](https://github.com/PalisadoesFoundation/talawa/blob/develop/.github/workflows/pull-request.yml) GitHub action is not failing on warnings. This means:

1. Tests are not sufficiently verifying code operation
1. Our code base is more prone to a poor UX.

Many of the tests that are failing are for widgets and the warnings imply screen overflow concerns. 

**To Reproduce**

## Method 1
Follow these steps:
1. Take a look at this [recent GitHub Action run](https://github.com/PalisadoesFoundation/talawa/actions/runs/4107719370/jobs/7087567772)
2. Click on the `Codebase testing` section and scroll to the various warnings

## Method 2
Follow these steps:
1. Run the `flutter test --coverage` command in the root directory of the repo

Here is some sample output:

```
Warning: A call to tap() with finder "exactly one widget with text "Next" (ignoring offstage widgets): Text("Next", inherit: true, color: Color(0xff008a37), size: 18.0, weight: 400, dependencies: [DefaultTextStyle, MediaQuery, _ScrollableScope])" derived an Offset (Offset(400.0, 676.1)) that would not hit test on the specified widget.
Maybe the widget is actually off-screen, or another widget is obscuring it, or the widget cannot receive pointer events.
Indeed, Offset(400.0, 676.1) is outside the bounds of the root of the render tree, Size(800.0, 600.0).
The finder corresponds to this RenderBox: RenderParagraph#298a5 relayoutBoundary=up3
The hit test result at that offset is: HitTestResult(HitTestEntry<HitTestTarget>#d3eb1(RenderView#d9235), HitTestEntry<HitTestTarget>#25b59(<AutomatedTestWidgetsFlutterBinding>))
#0      WidgetController._getElementPoint (package:flutter_test/src/controller.dart:956:25)
#1      WidgetController.getCenter (package:flutter_test/src/controller.dart:839:12)
#2      WidgetController.tap (package:flutter_test/src/controller.dart:273:18)
#3      main.<anonymous closure> (file:///home/runner/work/talawa/talawa/test/widget_tests/pre_auth_screens/signup_details_test.dart:205:18)
<asynchronous suspension>
#4      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>
To silence this warning, pass "warnIfMissed: false" to "tap()".
To make this warning fatal, set WidgetController.hitTestWarningShouldBeFatal to true.
```

**Expected behavior**
1. Tests fail on warning both from the CLI and in the GitHub action
1. Existing tests that fail with warnings must be fixed. The affected files must attain 100% test code coverage.
1. The overall test code coverage must not decline

**Actual behavior**
See above

**Screenshots**

image

**Additional details**
N/A

### PR DESCRIPTION
**Pull request by Azad99-9**
**What kind of change does this PR introduce?**

This PR makes `flutter test warnings` fatal.

**Issue Number:**

Fixes #1467 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

- Added a new file `test/flutter_test_config.dart`.
- This file gets executed `prior` to every test file in test folder, so that devs can configure test parameters at a single place.
- Added  `WidgetController.hitTestWarningShouldBeFatal = true` which makes all hitTest warnings fatal.
- Fixed all the `existing warnings` in the tests.

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Azad99-9** [comment]
@Cioppolo14 please assign reviewers for this PR.

File: lib/views/after_auth_screens/events/explore_event_dialogue.dart [hunk 1]
```diff
@@ -22,153 +19,135 @@ class _ExploreEventDialogState extends State<ExploreEventDialog> {
...
-      actions: [
-        Padding(
-          padding: const EdgeInsets.all(26),
-          child: Row(
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: [
-              Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
+      insetPadding:
+          EdgeInsets.symmetric(horizontal: SizeConfig.screenWidth! * 0.15),
```
Comment (literalEval) [inline lib/views/after_auth_screens/events/explore_event_dialogue.dart:23]: Why remove the trailing comma ?

File: lib/views/pre_auth_screens/set_url.dart [hunk 2]
```diff
@@ -37,189 +37,193 @@ class _SetUrlState extends State<SetUrl> {
...
+                            bottom: SizeConfig.safeBlockVertical! * 2,
+                          ),
+                          child: Row(
+                            mainAxisAlignment: MainAxisAlignment.center,
+                            children: [
+                              CustomPaint(
+                                size: Size(
+                                  SizeConfig.screenWidth! * 0.125,
+                                  SizeConfig.screenWidth! * 0.125 * 0.5,
+                                ), //You can Replace [WIDTH] with your desired width for Custom Paint and height will be calculated automatically
```
Comment (literalEval) [inline lib/views/pre_auth_screens/set_url.dart:202]: Remove these comments as part of this PR itself. They serve no purpose.

**Review by literalEval — CHANGES_REQUESTED** [review]
1. Are the extra files modified so that their tests can be changed too ?
2. Sometimes the warning comes from an old package or about something which we can't immediately fix. How to proceed in that case ?

**Comment by Azad99-9** [comment]
@literalEval 

> Are the extra files modified so that their tests can be changed too ?

I didn't get you kindly elaborate.

> Sometimes the warning comes from an old package or about something which we can't immediately fix. How to proceed in that case ?

- At the moment all the warnings previously existed were related to `hitTests` so I have added `WidgetController.hitTestWarningShouldBeFatal = true`. Additionally I have added a new file `flutter_test_config` which can be configured for all the warnings that are supported by `flutter testing framework`. 
- For the warnings which are `unforeseen` or `can't be fixed immediately` we can define a GitHub action However this won't work when we run test in CLI using `flutter test` which is also a requirement in the issue this PR is targetting. I am open to any solution from your side that covers this unexpected situation.

**Comment by Azad99-9** [comment]
@literalEval 

> I meant that you have changed some files and their tests too. It is so because these files were generating warnings earlier ?

- Yes, these files were generating warnings earlier because their UI wasn't `responsive`.
- As the `MediaQuery` constraints are `tight` in `test environment` this caused some UI elements to `exceed the screen` and when we try to `tap` these overflowed widgets the `hitTest` warnings were being generated.
- So I have made the repective UI components responsive while `preserving the existing functionality` and modified the tests to adhere to these new changes.

**Comment by Azad99-9** [comment]
> > Are the extra files modified so that their tests can be changed too ?
> 
> I meant that you have changed some files and their tests too. It is so because these files were generating warnings earlier ?
> 
> About the 'not instantly fixable warnings, I think if we fix the existing warnings for now, we may afford to add new features with full coverage ? I mean, say if we install a new package that doesn't have full test support, or a warning in itself, then we might choose to install another package that does the same job ? Though all of the major and popular packages have 100% coverage, there's always a culprit. And that's the concern here.

I think choosing an alternative in such situations would be a reasonable solution and I agree with this. coming to the existing warnings they are fixed in this PR without altering the existing functionality.

**Comment by Azad99-9** [comment]
@literalEval one instance of such UI components mentioned in [this](https://github.com/PalisadoesFoundation/talawa/pull/2332#issuecomment-1899857776) comment. Can be seen [here](https://github.com/PalisadoesFoundation/talawa/issues/2336#issue-2090310740).

**Comment by Azad99-9** [comment]
@literalEval kindly review.

**Review by literalEval — APPROVED** [review]
LGTM

**Review by noman2002 — APPROVED** [review]

**Pull request merged**